cloudinit/sources/DataSourceAzure.py

```diff
@@ -1399,34 +1704,71 @@ def read_azure_ovf(contents):
     if 'ssh_pwauth' not in cfg and password:
         cfg['ssh_pwauth'] = True
 
-    cfg['PreprovisionedVm'] = _extract_preprovisioned_vm_setting(dom)
+    preprovisioning_cfg = _get_preprovisioning_cfgs(dom)
+    cfg = util.mergemanydict([cfg, preprovisioning_cfg])
 
     return (md, ud, cfg)
 
 
 @azure_ds_telemetry_reporter
-def _extract_preprovisioned_vm_setting(dom):
-    """Read the preprovision flag from the ovf. It should not
-       exist unless true."""
+def _get_preprovisioning_cfgs(dom):
+    """Read the preprovisioning related flags from ovf and populates a dict
+    with the info."""
+    cfg = {}
+    cfg["PreprovisionedVm"] = False
+    cfg["PreprovisionedVMType"] = None
+
     platform_settings_section = find_child(
         dom.documentElement,
         lambda n: n.localName == "PlatformSettingsSection")
     if not platform_settings_section or len(platform_settings_section) == 0:
         LOG.debug("PlatformSettingsSection not found")
-        return False
+        return cfg
     platform_settings = find_child(
         platform_settings_section[0],
         lambda n: n.localName == "PlatformSettings")
     if not platform_settings or len(platform_settings) == 0:
         LOG.debug("PlatformSettings not found")
-        return False
+        return cfg
+
+    # Read the PreprovisionedVm bool flag. This should be deprecated when the
+    # platform has removed PreprovisionedVm and only surfaces
+    # PreprovisionedVMType.
     preprovisionedVm = find_child(
         platform_settings[0],
         lambda n: n.localName == "PreprovisionedVm")
     if not preprovisionedVm or len(preprovisionedVm) == 0:
         LOG.debug("PreprovisionedVm not found")
-        return False
-    return util.translate_bool(preprovisionedVm[0].firstChild.nodeValue)
+        return cfg
```

Comment:
In the future, when the platform has removed `PreprovisionedVm` and only surfaces `PreprovisionedVMType`, returning here means that the VM will not undergo the PPS Save Restore workflow. This decreases PPS Save Restore coverage.

Reply:
I meant to do it before but must have forgotten it. Refactored the logic to get these two values in two separate functions.